doc/ref_arch/kubernetes/chapters/chapter02.md

```diff
@@ -25,15 +25,131 @@
+
+Containerised Infrastructure Manager (CIM): a functional block that is responsible for controlling and managing the containerised NFVI compute, storage and network resources. Note CIM supports the Life Cycle Management of conatinerised and the underlying physical resources.
 
 <a name="2.3.1"></a>
 ### 2.3.1 General
 
 | Ref # | sub-category | Description |
 |----|------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| `req.gen.ost.01` | Open source | The Architecture **must** use Kubernetes APIs.|
-| `req.gen.ost.02` | Open source | The Architecture **must** support dynamic request and configuration of virtual resources (compute, network, storage) through Kubernetes APIs. |
+| `req.gen.k8s.01` | Open source | The Architecture **must** use Kubernetes APIs.|
+| `req.gen.k8s.02` | Open source | The Architecture **must** support dynamic request and configuration of resources (compute, network, storage) through Kubernetes APIs. |
+| `req.gen.cnt.01` | Cloud nativeness | The Architecture **should** consist of stateless service components. However, where state is required it must be kept external to the component. |
+| `req.gen.cnt.02` | Cloud nativeness | The Architecture **should** consist of service components implemented as microservices that are individually dynamically scalable. |
+| `req.gen.scl.01` | Scalability | The Architecture **should** support policy driven horizontal auto-scaling. |
+| `req.gen.rsl.01` | Resiliency | The Architecture **must** support resilient Kubernetes components that are required for the continued availability of running workloads. |
+| `req.gen.rsl.02` | Resiliency | The Architecture **should** support resilient Kubernetes service components that are not subject to `req.gen.rsl.01`. |
+| `req.gen.avl.01` | Availability | The Architecture **must** provide High Availability for Kubernetes components. |
+
+
+<p align="center"><b>Table 2-1:</b> Containerised Platform: General Requirements.</p>
+
+<a name="2.3.2"></a>
+### 2.3.2 Infrastructure
+
+| Ref # | sub-category | Description |
+|----|------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `req.inf.com.01` | Compute | The Architecture **must** provide compute resources for Pods.  |
+| `req.inf.com.02` | Compute | The Architecture **should** include industry standard hardware management systems at both HW device level (embedded) and HW platform level (external to device). |
+| `req.inf.stg.01` | Storage | The Architecture **must** provide shared Block storage for Containers.
+| `req.inf.stg.02` | Storage | The Architecture **must** provide shared Object storage for Containers.
```

Comment:
What do you mean by this?

Reply:
I don't think this is relevant - object storage isn't something that containers would use for Volumes, it is something the containerised applications would access over http(s). I don't think it's needed here.

Reply:
If we change the description text to 
> The architecture **must** provide shared object storage, for use by containerized workloads.
does that sufficiently break the perception of object storage being used as the basis for a volume?

Reply:
It does, but I'm not sure why the architecture has to provide shared object storage.  If consensus is that it does, what about other shared services such as DBaaS, etc.?  Where do we draw the line?

Reply:
I think for now we draw the line at (near?) parity with VM-based architectures, where the model and architecture are saying Object storage is supported.  DBaaS, etc are not in scope in that context.

Reply:
Agree with @kedmison. We should also remember that objects are useful abstractions and hence the need to support object storage -- totally abstracted from implementation details.

Reply:
Interesting - I'm surprised object storage is included in the VM-based architecture tbh and still not convinced we should include it here.

Reply:
Also, I can't find object storage listed in the RM anywhere (I do recall early discussions about whether to include it and the consensus was no). I would suggest,  given it's not really in scope of the RM, that we remove it from both RAs.

Reply:
@tomkivlin RM Section 5.2.2, '

.conf | Feature | Type | Basic | Network Intensive | Compute Intensive
-- | -- | -- | -- | -- | --
nfvi.stg.cfg.003 | Storage Object | Yes/No | Y | Y | Y

Reply:
Ah thanks - I was looking in chapters 3 and 4.  I'm still not convinced object storage belongs in CNTT, but hey ho.  Let's discuss on Thursday.

Reply:
@tomkivlin @pgoyal01 , I am with @tomkivlin  , I believe there is no object storage implementation, requirements from any VNF ,

Reply:
To be clear, I'm not saying that VNFs (or any other applications that might end up being in scope) don't have a requirement for object storage.  I'm saying that it's not an infrastructure service and therefore shouldn't be in scope of CNTT (in my opinion) - it's no different, to me, than DBaaS, or MQaaS, or any other foundational service that an application might choose to consume as a service.

Reply:
@tomkivlin we need to differentiate between the services that consumed by the infrastructure itself like DBaaS, or MQaaS, and services consumed by app/VNF/CNF itself like object or block storage 

in this context, we want to cover the infrastructure req. that should be exposed to CNF so my opinion to put it "Shall"

Reply:
Yes - I was referring to DB and MQ as required by the apps, not the infra.  I don't see object storage as being an infrastructure thing but let's take it through and see how it goes.  There's no difference between "must" and "shall" so perhaps make it a "should"?